### ISSUE TITLE
Stepout doesn't preserve current goroutine

### ISSUE THREAD

**Issue by nd** [go-delve/delve#2113]
1. What version of Delve are you using (`dlv version`)?

Version: 1.4.1
Build: f9c8f7f55b798a472013dd6a20bf688387eb0c53

2. What version of Go are you using? (`go version`)?

go1.15rc1

3. What operating system and processor architecture are you using?

linux/amd64

4. What did you do?

Debugging slightly modified version of https://github.com/aarzilli/delve_client_testing/blob/master/08gswitch/main.go:

```
package main

import (
	"fmt"
	"sync"
	"runtime"
)

func coroutine(i int, start, finish *sync.WaitGroup) {
	defer finish.Done()

	j := i * 2

	if i == 99 {
		runtime.Breakpoint()
		start.Done()
	} else {
		start.Wait()
  }

	fmt.Println("hello ", i, j)
	fmt.Println("goodbye", i, j)
}

func main() {
	i := 0
	var start, finish sync.WaitGroup
	start.Add(1)
	for ; i < 100; i++ {
		finish.Add(1)
		go coroutine(i, &start, &finish)
	}
	finish.Wait()
	println(i)
}
```

Once a breakpoint is reached, switched to another goroutine with main.coroutine in its stack. Stepped out.

5. What did you expect to see?

Stepout preserves the goroutine I've switched to and returns a state where current thread runs the selected goroutine.

6. What did you see instead?

Step out returns the current thread with a different goroutine (0 instead 6). 

This doesn't always happen, sometimes stepout returns the expected result.

Full log: https://gist.github.com/nd/038386b9fbdfe306753cdf17127fdb43.

**Comment by aarzilli** [comment]
How do you select the goroutine to switch to? Is it always 6? Is it always a goroutine running on a thread?

**Comment by aarzilli** [comment]
PS. what's the base rate of failure that I should expect?

**Comment by nd** [comment]
> How do you select the goroutine to switch to?

Listed goroutines, selected the first one containing main.coroutine in the stack. I've added shuffle for goroutines before checking the stack, so goroutine is random every time - the test is still fails, usually within 100 runs. In failures I've seen the goroutine I've switched to was always running on a thread: the state returned when breakpoint is reached contains this goroutineId in one of the threads.
